packages/hash/frontend/src/components/GlideGlid/glide-grid.tsx

```diff
@@ -62,6 +88,32 @@ const GlideGrid: ForwardRefRenderFunction<DataEditorRef, GlideGridProps> = (
     return () => InteractableManager.deleteInteractables(tableId);
   }, []);
 
+  const [propertySort, setPropertySort] = useState<TableSort<string>>(
+    initialPropertySort ?? {
+      key: columns[0]?.id ?? "",
+      dir: "asc",
+    },
+  );
+
+  const defaultDrawHeader = useDrawHeader(
+    sortable ? propertySort : undefined,
+    columns,
+  );
+
+  const handleHeaderClicked = sortable
+    ? createHandleHeaderClicked(columns, propertySort, setPropertySort)
+    : undefined;
+
+  const rows = useMemo(() => {
+    if (!sortable) {
+      return rowData;
+    }
+
+    const sortedRows = sortRowData(rowData, propertySort);
+
+    return onSort?.(sortedRows) ?? sortedRows;
```

Comment:
I think the API here could be clearer if instead of having two different processes here, you just have `sortRowData` as a prop which defaults to this implementation, but can be overwritten to wrap this implementation of `sortRowData` to, for example, flatten.